src/compiler/checker.ts

```diff
@@ -9543,6 +9543,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 initializer: Expression | undefined
             ) => T, methodKind: SignatureDeclaration["kind"], useAccessors: boolean): (p: Symbol, isStatic: boolean, baseType: Type | undefined) => (T | AccessorDeclaration | (T | AccessorDeclaration)[]) {
                 return function serializePropertySymbol(p: Symbol, isStatic: boolean, baseType: Type | undefined): (T | AccessorDeclaration | (T | AccessorDeclaration)[]) {
+                    if (methodKind === SyntaxKind.MethodDeclaration && getCheckFlags(p) & CheckFlags.Instantiated) {

```

Comment:
```suggestion
                    if (methodKind === SyntaxKind.MethodDeclaration) {
                        p = getTargetSymbol(p);
                    }
```
But I'm not sure this is 100% right in the first place - symbol instantiation is used for much more than getting rid of a `this` type - it's used for copying type parameters (as you can see in the other baseline change), for instantiating the parameter types of signatures, and generally for instantiating object type members.

Probably a better fix would be finding where we lookup the types/symbols to feed into the serializer at the layer that calls this one and make sure we're getting them w/o an instantiated `this` in the first place.